homeassistant/components/zwave/climate.py

```diff
@@ -131,21 +142,35 @@ def update_properties(self):
         if self.values.mode:
             self._hvac_list = []
             self._hvac_mapping = {}
+            self._preset_list = []
+            self._preset_mapping = {}
+
             hvac_list = self.values.mode.data_items
             if hvac_list:
                 for mode in hvac_list:
                     ha_mode = HVAC_STATE_MAPPINGS.get(mode)
+                    ha_preset = PRESET_MAPPING.get(mode)
                     if ha_mode and ha_mode not in self._hvac_mapping:
                         self._hvac_mapping[ha_mode] = mode
                         self._hvac_list.append(ha_mode)
-                        continue
-                    self._hvac_list.append(mode)
+                    elif ha_preset and ha_preset not in self._preset_mapping:
+                        self._preset_mapping[ha_preset] = mode
+                        self._preset_list.append(ha_preset)
+                    else:
+                        # If nothing matches
+                        self._hvac_list.append(mode)
```

Comment:
Dear Reviewer, do you think it's better to add unmatching modes to the hvac_list, or should these modes be added to the preset_list?

Reply:
@Santobert , I've been following your work.  I've tried a few different things to solve my problem, but again, with little programming experience I haven't gotten very far.  

From reviewing this on the developers site (https://developers.home-assistant.io/docs/en/entity_climate.html), I think it would be best to append to the preset_list.  My thought is that building it as a preset would do two things:

1. Ensure other zwave climate devices that have unique operating modes don't have blank entries for hvac_mode in the front end.
2. Allow operating modes from other zwave climate devices to work without having to enter the specific modes for every zwave device.

**From the developers docs:**
"You are only allowed to use the built-in HVAC modes. If you want another mode, add a preset instead."
"A device can have different presets that it might want to show to the user. Common presets are "Away" or "Eco". There are a couple of built-in presets that will offer translations, but you're also allowed to add custom presets."

If this doesn't make sense, please disregard.  Again, I'm not a programmer nor do I have intimate knowledge of the inner workings of home assistant.  Just throwing my 2 cents out there.

Reply:
I agree. Thank you for your opinion